dfpl/single_label_model.py

```diff
@@ -18,27 +23,35 @@
 from sklearn.model_selection import StratifiedKFold, train_test_split
 from tensorflow.keras import metrics, optimizers, regularizers
 from tensorflow.keras.layers import AlphaDropout, Dense, Dropout
-from tensorflow.keras.losses import BinaryCrossentropy, MeanSquaredError
+from tensorflow.keras.losses import (
+    BinaryCrossentropy,
+    MeanSquaredError,
+    BinaryFocalCrossentropy,
+)
 from tensorflow.keras.models import Model, Sequential
 
 from dfpl import callbacks as cb
 from dfpl import options
 from dfpl import plot as pl
 from dfpl import settings
+from dfpl.utils import ae_scaffold_split, weight_split
 
 
 def prepare_nn_training_data(
-    df: pd.DataFrame, target: str, opts: options.Options
-) -> (np.ndarray, np.ndarray):
-    # check the value counts and abort if too imbalanced
+    df: pd.DataFrame, target: str, opts: options.Options, return_dataframe: bool = False
+) -> Union[Tuple[np.ndarray, np.ndarray], pd.DataFrame]:
```

Comment:
This makes it a bit hard to use because one always needs to check if a DataFrame or a tuple has been returned.

Reply:
Also the function is not used correctly in vae.py: the use assumes a triple (and there is no seed parameter)

```
train_data, val_data, test_data = weight_split(
                df, sizes=(1 - opts.testSize, 0.0, opts.testSize), bias="small", seed=42
            )
```

wondering if this works at all. 

Unfortunatelly you completely ignored my request for a test.

Reply:
In order to find such problems easily you can use `mypy dfpl` (note that some errors predate your PR :) )

Reply:
> This makes it a bit hard to use because one always needs to check if a DataFrame or a tuple has been returned.

Well the retun_dataframe argument solves this ambiguity,no?



> Also the function is not used correctly in vae.py: the use assumes a triple (and there is no seed parameter)
> 
> ```
> train_data, val_data, test_data = weight_split(
>                 df, sizes=(1 - opts.testSize, 0.0, opts.testSize), bias="small", seed=42
>             )
> ```
> 
> wondering if this works at all.
> 
> Unfortunately, you completely ignored my request for a test.

I didnt know about mypy. On it!

Reply:
> Well the retun_dataframe argument solves this ambiguity,no?

Indeed

Regarding `weight_split` .. somehow I mixed up the functions .. but the comment regarding mypy is still valid. But I think you only need to fix errors concerning your contributions.